include/micm/cuda/util/cuda_util.cuh

```diff
@@ -30,5 +33,54 @@ namespace micm
 
     /// @brief Get the cuBLAS handle for the current device
     cublasHandle_t& GetCublasHandle();
+
+    /// @brief Define a functor for the cudaStream unique pointer deleter
+    struct CudaStreamDeleter
+    {
+      void operator()(cudaStream_t* cuda_stream) const
+      {
+        if (cuda_stream != nullptr)
+        {
+          CHECK_CUDA_ERROR(cudaStreamDestroy(*cuda_stream), "CUDA stream finalization failed");
+          delete cuda_stream;
+          cuda_stream = nullptr;
+        }
+      }
+    };
+
+    /// @brief Define the smart pointer type using the functor for the custom deleter
+    using CudaStreamPtr = std::unique_ptr<cudaStream_t, CudaStreamDeleter>;
+
+    /// @brief Singleton class to manage CUDA streams
+    class CudaStreamSingleton
+    {
+    public:
+
+      CudaStreamSingleton() = default;
```

Comment:
I believe this should be `private` in a singleton implementation

Reply:
Thanks for catching it. Fixed.